File: Domains/Data Structures/Trees/Binary Search Tree - Insertion/Insertion.cc
/*
Node is defined as

typedef struct node
{
   int data;
   node * left;
   node * right;
}node;

*/


node * insert(node* root, int value)
{
  if(!root)
  {
    root = new node;
    root->left = root->right = NULL;
    root->data = value;
  }
  else
  {
    if(value > root->data) root->right = insert(root->right, value);
    else root->left = insert(root->left, value);
  }
  return root;
}

File: Domains/Data Structures/Trees/Binary Search Tree - Insertion/Insertion.java
/* Node is defined as :
class Node
   int data;
   Node left;
   Node right;

   */

static Node Insert(Node root,int value)
   {
     if(root == null)
     {
       Node temp = new Node();
       temp.data = value;
       temp.right = temp.left = null;
       return temp;
     }
     else
     {
       if(value > root.data) root.right = Insert(root.right, value);
       else root.left = Insert(root.left, value);
     }
     return root;
   }
